app/api/routes.py

```diff
@@ -72,3 +80,29 @@ def get_languages():
         language_list = []
     finally:
         return jsonify(language_list)
+
+
+def set_resource(id, param_list):
+    resource = None
+    try:
+        resource = Resource.query.get(id)
```

Comment:
While I agree it might be nice to abstract out, after looking over the code for updating/creating a resource in the CLI I am afraid that it might not be wise to try abstracting out the code since the CLI resources are pulled from the current DB session, with all their field values stored in dynamic dictionaries rather than setting explicit values passed by users.

There is also the issue with how the CLI resources only have some of their fields ('category' and 'languages') updated, which makes refactoring difficult since in this case all resource fields are being updated with information from user input. For now I will just edit the Route code to have it properly update/create a resource while leaving the CLI code as it is.

Reply:
Yeah, those are good points, perhaps we can't really share that code easily